packages/devtools_app/lib/src/code_size/code_size_controller.dart

```diff
@@ -82,113 +71,119 @@ class CodeSizeController {
     changeRoot(newRoot);
   }
 
-  /// Builds a tree with [TreemapNode] from [treeJson] which represents
-  /// the hierarchical structure of the tree.
   TreemapNode generateTree(Map<String, dynamic> treeJson) {
-    var treemapNodeName = treeJson['n'];
-    if (treemapNodeName == '') treemapNodeName = 'Unnamed';
-    final rawChildren = treeJson['children'];
-    final treemapNodeChildren = <TreemapNode>[];
-
-    int treemapNodeSize = 0;
-    if (rawChildren != null) {
-      // If not a leaf node, build all children then take the sum of the
-      // children's sizes as its own size.
-      for (dynamic child in rawChildren) {
-        final childTreemapNode = generateTree(child);
-        if (childTreemapNode == null) {
-          continue;
-        }
-        treemapNodeChildren.add(childTreemapNode);
-        treemapNodeSize += childTreemapNode.byteSize;
-      }
+    final isLeafNode = treeJson['children'] == null;
+    if (!isLeafNode) {
+      return buildNodeWithChildren(treeJson);
     } else {
-      // If a leaf node, just take its own size.
-      treemapNodeSize = treeJson['value'];
-
-      // Only add nodes with a size.
-      if (treemapNodeSize == null || treemapNodeSize == 0) {
+      // TODO(peterdjlee): Investigate why there are leaf nodes with size of null.
+      final byteSize = treeJson['value'];
+      if (byteSize == null) {
         return null;
       }
+      return buildNode(treeJson, byteSize);
     }
-
-    final childrenMap = <String, TreemapNode>{};
-
-    for (TreemapNode child in treemapNodeChildren) {
-      childrenMap[child.name] = child;
-    }
-
-    return TreemapNode(
-      name: treemapNodeName,
-      byteSize: treemapNodeSize,
-      childrenMap: childrenMap,
-    )..addAllChildren(treemapNodeChildren);
   }
 
-  /// Builds a tree with [TreemapNode] from [treeJson] which represents
-  /// the hierarchical structure of the tree.
+  /// Recursively generates a diff tree from [treeJson] that contains the difference
+  /// between an old snapshot and a new snapshot. Each node in the resulting tree
```

Comment:
for dart doc, the first sentence should be separated from all others by a new line in the comment

Reply:
Done